rasa/cli/data.py

```diff
@@ -234,6 +273,422 @@ def split_nlu_data(args: argparse.Namespace) -> None:
+    ):
+        if intent not in classification_report.keys():
+            continue
+
+        intent_results_original = classification_report[intent]
+        intent_results = intent_report[intent]
+
+        # Record performance changes for augmentation based on the diversity criterion
+        precision_change = (
+            intent_results["precision"] - intent_results_original["precision"]
+        )
+        recall_change = intent_results["recall"] - intent_results_original["recall"]
+        f1_change = intent_results["f1-score"] - intent_results_original["f1-score"]
+
+        intent_results["precision_change"] = intent_summary[intent][
+            "precision_change"
+        ] = precision_change
+        intent_results["recall_change"] = intent_summary[intent][
+            "recall_change"
+        ] = recall_change
+        intent_results["f1-score_change"] = intent_summary[intent][
+            "f1-score_change"
+        ] = f1_change
+        intent_report[intent] = intent_results
+
+    return (intent_summary, intent_report)
+
+
+def _plot_summary_reports(
+    intent_summary_diverse: Dict[Text, Dict[Text, float]],
+    intent_summary_random: Dict[Text, Dict[Text, float]],
+    changed_intents_diverse: Set[Text],
+    changed_intents_random: Set[Text],
+    output_directory_diverse: Text,
+    output_directory_random: Text,
+):
+
+    for metric in ["precision", "recall", "f1-score"]:
+        output_file_diverse = os.path.join(
+            output_directory_diverse, f"{metric}_changes.png"
+        )
+        rasa.utils.plotting.plot_intent_augmentation_summary(
+            augmentation_summary=intent_summary_diverse,
+            changed_intents=changed_intents_diverse,
+            metric=metric,
+            output_file=output_file_diverse,
+        )
+
+        output_file_random = os.path.join(
+            output_directory_random, f"{metric}_changes.png"
+        )
+        rasa.utils.plotting.plot_intent_augmentation_summary(
+            augmentation_summary=intent_summary_random,
+            changed_intents=changed_intents_random,
+            metric=metric,
+            output_file=output_file_random,
+        )
+
+
+def suggest_nlu_data(args: argparse.Namespace) -> None:
+    """Load NLU training & evaluation data, paraphrases, an existing classification report, its corresponding config
+    file and suggest additional training
+     examples.
+
+    Args:
+        args: Commandline arguments
+    """
+    nlu_training_data = rasa.shared.nlu.training_data.loading.load_data(
+        args.nlu_training_data
+    )
+    nlu_evaluation_data = rasa.shared.nlu.training_data.loading.load_data(
+        args.nlu_evaluation_data
+    )
+    paraphrases = rasa.shared.nlu.training_data.loading.load_data(args.paraphrases)
+    classification_report = rasa.shared.utils.io.read_json_file(
+        args.nlu_classification_report
+    )
+    random.seed(args.random_seed)
```

Comment:
We're calling `train_nlu(...)` inside the cli command, so it seems likely that there'll be another call to `random.seed(...)` somewhere. What would be the best way to deal with it? Adding the random seed to the config when calling `train_nlu(...)`?

Reply:
That would work!

Reply:
Right, this turns out to be fairly fiddly. We're only passing on a path to the `config` file until calling `train_nlu` and it seems overkill to load the config, add the random seed (or check whether one is already set - what if there is a clash?), and write it back to file.

What we want to achieve with random seeds is basically reproducibility, and I think it might be best to set the random seed in the data augmentation functionality right before it is used. Subsequent changes e.g. in `train_nlu(...)` are fine and ensure that whatever is set in the config file is reproducible as well. In that sense, its probably best to set the random seed in data augmentation right before its usage and allow subsequent calls to change the seed again.